gradio/components.py

```diff
@@ -3965,8 +3967,296 @@ def postprocess(self, y: str | None) -> Dict[str, str] | None:
             out_y = y.to_json()
         return {"type": dtype, "plot": out_y}
 
-    def style(self):
-        return self
+    def style(self, container: Optional[bool] = None):
+        return IOComponent.style(
+            self,
+            container=container,
+        )
+
+
+@document("change", "clear")
+class ScatterPlot(Plot):
+    """
+    Create a scatter plot.
+
+    Preprocessing: this component does *not* accept input.
+    Postprocessing: expects a pandas dataframe with the data to plot.
+
+    Demos: native_plots
+    """
+
+    def __init__(
+        self,
+        x: Optional[str] = None,
+        y: Optional[str] = None,
+        value: Optional[pd.DataFrame] = None,
```

Comment:
I would have expected the `value` be the first parameter (that's what's in the proposed API in the PR) but also it's similar to how other plotting libraries do it. Suggest reordering? 

Also what happens if `x` and `y` are not provided? Should we throw an error or is there an automatic way to infer them? 

```suggestion
        value: Optional[pd.DataFrame] = None,
        x: Optional[str] = None,
        y: Optional[str] = None,
        *
```